src/main/java/com/google/cloud/discotoproto3converter/proto3/DocumentToProtoConverter.java

```diff
@@ -314,15 +316,28 @@ private void readResources(Document document) {
           httpOptionPath =
               httpOptionPath.replace(
                   "{" + pathParam.getIdentifier() + "}", "{" + pathField.getName() + "}");
+          
+          // required method signatures
+          methodSignatureParams.add(pathField.getName());
         }
 
+        Option requiredMethodSignatureOption = new Option("google.api.method_signature");
+        requiredMethodSignatureOption.getProperties().put("", String.join(",", methodSignatureParams));
+
         for (Schema queryParam : method.queryParams().values()) {
           Field queryField = schemaToField(queryParam);
           input.getFields().add(queryField);
           if (queryField.getValueType().isEnum()) {
             input.getEnums().add(queryField.getValueType());
           }
+
+          // optional method signatures
+          methodSignatureParams.add(queryField.getName());
         }
+
+        Option optionalMethodSignatureOption = new Option("google.api.method_signature");
```

Comment:
Let's have a plan for the optional signatures before we put in code to collect them. This new Option never seems to get read; let's not add dead code.

(You can insert a TODO, though, as a helpful pointer.)